### PR TITLE
Remove Germany and Customized clouds

### PR DESCRIPTION
**Pull request by kostrse**
Germany cloud has been deprecated some time ago.

Dedicated definition for "Customized" cloud also not needed because Azure SDK doesn't enforce constraints on passed names of clouds.

### CODE REVIEW

**Comment by kostrse** [comment]
@andresmgot

**Review by yuwaMSFT2 — APPROVED** [review]

File: azsettings/clouds.go [hunk 1]
```diff
@@ -6,8 +6,6 @@ const (
 	AzurePublic       = "AzureCloud"
 	AzureChina        = "AzureChinaCloud"
 	AzureUSGovernment = "AzureUSGovernment"
-	AzureGermany      = "AzureGermanCloud"
-	AzureCustomized   = "AzureCustomizedCloud"
```
Comment (andresmgot) [inline azsettings/clouds.go:10]: this is a breaking change, `grafana/grafana` is still using these exported variables, for example: https://github.com/grafana/grafana/blob/main/pkg/tsdb/azuremonitor/credentials.go#L55

Comment (yuwaMSFT2) [reply azsettings/clouds.go:10]: @kostrse may need to update all other places to remove the reference, as Germany cloud was removed last year (no such cloud any more).

Comment (kostrse) [reply azsettings/clouds.go:10]: The Germany cloud has been deprecated in 2020 closed on October 29. 2021 and cannot be used.
https://learn.microsoft.com/en-us/azure/active-directory/develop/authentication-national-cloud

All references to it in the code and UI can be removed in Grafana and ADX with a release note.

Users should not be impacted.

Comment (yuwaMSFT2) [reply azsettings/clouds.go:10]: Thanks @andresmgot ! 
qq: is Grafana core always picking up the latest SDK version?
@kostrse let's work with @andresmgot to learn on the version control and follow the best practice on release.

**Comment by andresmgot** [comment]
@kostrse can you open a PR for grafana to update there the SDK version and fix the breaking changes?

**Comment by kostrse** [comment]
Azure Monitor and Prometheus have drop down option for Germany. This options doesn't work since this cloud doesn't exist, so customers won't be affected if we remove Germany option from Azure Monitor and Prometheus.

I can send PR and have it targeted for 9.3. Would it be good to remove mentions of Germany in 9.3?

**Comment by andresmgot** [comment]
yes, that would be fine

**Comment by ryantxu** [comment]
See https://github.com/grafana/grafana/pull/60362

**Pull request merged**